code/modules/projectiles/guns/smgs.dm

```diff
@@ -353,3 +353,82 @@
 	scatter_unwielded = 4
 	aim_slowdown = 0.15
 	wield_delay = 0.2 SECONDS
+
+
+//-------------------------------------------------------
+//Vector, based on KRISS vector 45ACP.
+/obj/item/weapon/gun/smg/standard_stormsmg
+	name = "\improper SMG-45 storm submachinegun"
+	desc = "The SMG-45 is the TerraGov Marine Corps depelopment to increase assault capability of marines. Lightweight and simple to use. It features delayed blowback system, heavily reducing recoil even with its high ROF. A highly-customizable platform, it is reliable and versatile. Ideal weapon for quick assaults. Uses extended .45 ACP magazines"
+	fire_sound = 'sound/weapons/guns/fire/tp23.ogg'
+	icon = 'icons/Marine/gun64.dmi'
+	icon_state = "v45"
+	item_state = "v45"
+	caliber = CALIBER_45ACP //codex
+	max_shells = 25 //codex
+	flags_equip_slot = ITEM_SLOT_BACK
+	force = 20
+	type_of_casings = null
+	//default_ammo_type = /obj/item/ammo_magazine/pistol/standard_heavypistol
+	//allowed_ammo_types = list(/obj/item/ammo_magazine/pistol/standard_heavypistol)
+	default_ammo_type = /obj/item/ammo_magazine/smg/standard_stormsmg
+	allowed_ammo_types = list(
+		/obj/item/ammo_magazine/smg/standard_stormsmg,
+		/obj/item/ammo_magazine/smg/standard_stormsmg/drum,
+	)
+	attachable_allowed = list(
```

Comment:
a SMG with that much attachable is not an smg...

like, scopes ? bipod ???? flamer and grenade launcher ????? 
It should have the same attachables than a smg90